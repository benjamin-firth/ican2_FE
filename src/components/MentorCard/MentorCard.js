import React from 'react';
import { Link } from 'react-router-dom';
import './MentorCard.scss';

const MentorCard = ({ mentor }) => {

  return (
      <section className='mentor-card-container'>
        <img
          src={mentor.profile.image}
          alt='mentor profile picture'
          className='profile-photo'/>
        <h2>{mentor.name.toLowerCase()}</h2>
        <h3>{mentor.mentorProfile.fieldOfKnowledge}</h3>
        {mentor.profile.gender !== 'none-specified' ? <p>{mentor.location.city}, {mentor.location.state} | {mentor.profile.gender}</p> : <p>{mentor.location.city}, {mentor.location.state}</p>}
        <p>{mentor.profile.aboutMe}</p>
        <Link to={`/mentors/${mentor.id}`}>
          <button>view profile</button>
        </Link>
      </section>
  );
}

export default MentorCard;
