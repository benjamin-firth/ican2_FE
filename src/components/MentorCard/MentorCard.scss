@import '../../variables.scss';

.mentor-card-container {
  margin: 2vw;
  padding: 2vw;
  height: 30vw;
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 25vw;
  align-items: center;
  background-color: white;
  button {
    border: 5px solid $secondarycolor;
  }
  img {
    border-radius: 50%;
    width: 12vw;
    margin-bottom: 3vw;
  }
  h2, h3, p {
    margin: 0;
    text-align: center;
  }
  h2, h3 {
    margin-bottom: .5vw;
  }
}
