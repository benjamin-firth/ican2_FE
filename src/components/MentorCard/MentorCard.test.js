import React from 'react';
import { shallow } from 'enzyme';
import MentorCard from './MentorCard';

describe('MentorCard', () => {
  let wrapper;
  let mockMentor = {
    id: 1,
    name: 'mockName',
    profile: {
      image: 'mockUrl',
      gender: 'He / Him',
      aboutMe: 'mock aboutMe'
    },
    mentorProfile: {
      fieldOfKnowledge: 'mock field'
    },
    location: {
      city: 'mockCity',
      state: 'mockState'
    }
  }

  let mockMentor2 = {
    id: 1,
    name: 'mockName',
    profile: {
      image: 'mockUrl',
      gender: 'none-specified',
      aboutMe: 'mock aboutMe'
    },
    mentorProfile: {
      fieldOfKnowledge: 'mock field'
    },
    location: {
      city: 'mockCity',
      state: 'mockState'
    }
  }

  it('should match the snapshot', () => {
    wrapper = shallow( <MentorCard mentor={mockMentor} />)

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when gender is no specified', () => {
    wrapper = shallow( <MentorCard mentor={mockMentor2} />)

    expect(wrapper).toMatchSnapshot();
  });

});
