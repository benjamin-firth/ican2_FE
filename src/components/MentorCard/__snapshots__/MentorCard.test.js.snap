// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MentorCard should match the snapshot 1`] = `
<section
  className="mentor-card-container"
>
  <img
    alt="mentor profile picture"
    className="profile-photo"
    src="mockUrl"
  />
  <h2>
    mockname
  </h2>
  <h3>
    mock field
  </h3>
  <p>
    mockCity
    , 
    mockState
     | 
    He / Him
  </p>
  <p>
    mock aboutMe
  </p>
  <Link
    to="/mentors/1"
  >
    <button>
      view profile
    </button>
  </Link>
</section>
`;

exports[`MentorCard should match the snapshot when gender is no specified 1`] = `
<section
  className="mentor-card-container"
>
  <img
    alt="mentor profile picture"
    className="profile-photo"
    src="mockUrl"
  />
  <h2>
    mockname
  </h2>
  <h3>
    mock field
  </h3>
  <p>
    mockCity
    , 
    mockState
  </p>
  <p>
    mock aboutMe
  </p>
  <Link
    to="/mentors/1"
  >
    <button>
      view profile
    </button>
  </Link>
</section>
`;
